src/providers/rust.rs

```diff
@@ -233,6 +218,141 @@ impl RustProvider {
+                // a member can have globs
+                if default_member.contains('*') || default_member.contains('?') {
+                    for member in app.find_directories(default_member)? {
+                        let mut manifest =
+                            app.read_toml::<Manifest>(&format!("{}/Cargo.toml", member.display()))?;
+
+                        manifest.complete_from_path(
+                            &app.source.join(format!("{}/Cargo.toml", member.display())),
+                        )?;
+
+                        if let Some(package) = manifest.package {
+                            if !manifest.bin.is_empty() || manifest.lib.is_none() {
+                                return Ok(Some(package.name));
+                            }
+                        }
+                    }
+                } else {
+                    let mut manifest =
+                        app.read_toml::<Manifest>(&format!("{}/Cargo.toml", default_member))?;
+
+                    manifest.complete_from_path(
+                        &app.source.join(format!("{}/Cargo.toml", default_member)),
+                    )?;
+
+                    if let Some(package) = manifest.package {
+                        if !manifest.bin.is_empty() || manifest.lib.is_none() {
+                            return Ok(Some(package.name));
+                        }
+                    }
+                }
+            }
+
+            for member in workspace
+                .members
+                .iter()
+                .filter(|member| !workspace.exclude.contains(member))
+            {
+                // a member can have globs
+                if member.contains('*') || member.contains('?') {
+                    for member in app.find_directories(member)? {
+                        let mut manifest =
+                            app.read_toml::<Manifest>(&format!("{}/Cargo.toml", member.display()))?;
+
+                        manifest.complete_from_path(
+                            &app.source.join(format!("{}/Cargo.toml", member.display())),
+                        )?;
+
+                        if let Some(package) = manifest.package {
+                            if !manifest.bin.is_empty() || manifest.lib.is_none() {
+                                return Ok(Some(package.name));
+                            }
+                        }
+                    }
+                } else {
+                    let mut manifest =
+                        app.read_toml::<Manifest>(&format!("{}/Cargo.toml", member))?;
+
+                    manifest
+                        .complete_from_path(&app.source.join(format!("{}/Cargo.toml", member)))?;
+
+                    if let Some(package) = manifest.package {
+                        if !manifest.bin.is_empty() || manifest.lib.is_none() {
+                            return Ok(Some(package.name));
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Sam with this code. It is reused from the default member stuff above. It could be abstracted into a function to remove the duplication.

Reply:
Done in https://github.com/railwayapp/nixpacks/pull/402/commits/7879e9f935bb57dabfa038b25ab6d6679df2d2cb